infor-design/enterprise-wc
# 2115 - IdsDataGrid failed tests fixes

**Explain the details for making this change. What existing problem does the pull request solve?**
Addresses bugs found in datagrid test conversion.
Re-enabled those skipped tests and refactored a couple.
Fixed dropdown style bug with `white-space: pre`, it should only apply to cells configured for text formats.

**Related github/jira issue (required)**:
Closes #2115 

**Steps necessary to review your pull request (required)**:
1. Check that test suite passes with re-enabled datagrid tests

**Test `activecellchanged` event**
1. Go to http://localhost:4300/ids-data-grid/example.html
2. Run `$('ids-data-grid').addEventListener('activecellchanged', (e) => console.log(e.detail.activeCell))`
3. Navigate cells via keyboard and check that `activecellchanged` event is output in console

**Test Date/Time filters**
1. Go to http://localhost:4300/ids-data-grid/filter-trigger-fields.html
2. Test `Pub. Date` , `Pub. Time` columns filters

**Test `is-not-empty` condition**
1. Go to http://localhost:4300/ids-data-grid/filter-trigger-fields.html
2. For columns `Price (Int)` or `Price`, set filter condition to `Is Not Empty`
3. Should correctly filter rows with populated cells

**Test `endCellEdit()`**
1. Go to http://localhost:4300/ids-data-grid/editable.html
2. Run `$('ids-data-grid').cellByIndex(0, 2).endCellEdit()`
3. Should no longer produce error

**Test hyperlink cells with `Enter`**
1. Go to http://localhost:4300/ids-data-grid/example.html#
2. Keyboard navigate to cell under `Location` column
3. Press `Enter`
4. Should see `Link was clicked` in console

**Test button cells with `Enter`**
1. Go to http://localhost:4300/ids-data-grid/columns-formatters.html
2. Keyboard navigate to cell under `Actions`
3. Press `Enter`
4. Should see `Actions clicked` in console

**Test bug fix introduced with `white-space: pre`**
1. Go http://localhost:4300/ids-data-grid/editable.html
2. Select value from 'Currency` dropdown
3. Text should be aligned correctly again

**Included in this Pull Request**:
